src/coreclr/jit/codegencommon.cpp

```diff
@@ -4177,7 +4176,7 @@ void CodeGen::genHomeSwiftStructParameters(bool handleStack)
             else
             {
                 var_types loadType = TYP_UNDEF;
-                switch (seg.Size)
+                switch (RISCV64_ONLY(RoundUpToPower2)(seg.Size))
```

Comment:
Can you just add the 3, 5, 6, 7 cases below? I'm adding them as part of #101372 anyway.